packages/@webex/plugin-meetings/src/multistream/receiveSlotManager.ts

```diff
@@ -71,6 +71,15 @@ export class ReceiveSlotManager {
       (csi: CSI) => this.meeting.members.findMemberByCsi(csi)?.id
     );
 
+    // update media requests on source update
+    if (getMediaFamily(mediaType) === MediaFamily.Video) {
+      receiveSlot.on(ReceiveSlotEvents.SourceUpdate, () => {
+        this.meeting.mediaRequestManagers[
```

Comment:
hmm... it's not ideal that ReceiveSlotManager now has a dependency on MediaRequestManager... maybe it would be better the other way around? ReceiveSlotManager doesn't know anything about media requests, but MediaRequestManager is already aware of receive slots (because they're part of media requests) so it would make more sense for MediaRequestManager to depend on ReceiveSlotManager and just listen on some event, maybe even MediaRequestManager can just listen for SourceUpdate events directly on the receive slots that are used in the media requests it manages?
Also I would rather avoid having the logic of doing this only for videos here in ReceiveSlotManager. I think that logic belongs in MediaRequestManager or could be configured for the right instance of MediaRequestManager by the meeting object.

Reply:
Yeah, I think it makes sense inside of `MediaRequestManager`. I'm thinking maybe I can listen to the event in `addRequest` and remove the listener in `cancelRequest`. The degradation logic already only degrades video requests, so I can probably remove the logic to check for the media family.

Reply:
yeah it's much better now, but I'm worried a bit about us sending excessive number of media requests now... I'll 1-1 with you about the details